src/lambda_function/dispatch/app.py

```diff
@@ -0,0 +1,141 @@
+"""
+Dispatch Lambda
+
+This Lambda polls the input-to-dispatch SQS queue and publishes to the dispatch SNS topic.
+Its purpose is to inspect each export and dispatch each file as a separate job in which
+the file will be decompressed and uploaded to S3.
+"""
+import json
+import logging
+import os
+import zipfile
+from urllib import parse
+
+import boto3
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def get_object_info(s3_event: dict) -> dict:
```

Comment:
nit: We could add something like a `utils.py` script, add common used functions like this one and have the lambdas (s3 to glue lambda also uses this) pull from this so we don't have to maintain redundant code and tests in two different places

Reply:
That's a good idea, but I'm skeptical we should pursue this because:
1. It adds complexity: From the [docs](https://docs.aws.amazon.com/lambda/latest/dg/python-package.html#python-package-dependencies), it sounds like we either need to maintain and deploy a separate python package and then use it as a layer -- which involves a lot of deployment/abstraction --  or we can include it as a separate file, which seems a bit easier, but maintaining that code in a single file yet including that file in both Lambda deployments involves a bit of sceptre-fu which is unclear to me how we could do that easily. Would a soft link work? I've never used that as part of git.
2. I eventually hope to replace the bucket which our S3 to JSON job reads from with the raw bucket, or potentially remove S3 to JSON altogether and immediately work with the data in Spark if possible. So at some point the input -> intermediate flow could be replaced and this logic would only be in one place.

Reply:
Ahhh.... of course it's more complicated for lambdas. I also took a brief look, according to this: https://aws.plainenglish.io/how-to-share-code-between-lambda-functions-49c656bd2ffc it doesn't seem like we need to deploy a separate python package just have a separate stack config for the shared layer containing the shared code, and have both lambda functions reference the layer and its arn output. Then we should be able to upload the code to our cf bucket and deploy via sceptre per usual.

But agreed the added complexity is that what we need right now, we should be looking for ways (like 2. as you mentioned) to reduce complexity of the pipeline. I'm for keeping this as is.